### PR TITLE
Adarsh int 1198 ft refactor

### PR DESCRIPTION
**Pull request by AVandana-OSI**
### Technical Description
Cleaned up the authentication classes and created the interface to get the token.  Moved the yaml config files to separate package.

If other projects want to use the same implementation to get the token for there related API's, they have to create a separate yml file in their own project and extend the ConfigReader, so api-core can read your yaml values. Then call AuthenticationUtils to get the token based on the user type. Ferb API has already implemented few test cases. 

### Reference
https://github.com/ca-cwds/API/blob/master/src/test/functional/java/gov/ca/cwds/api/FunctionalTest.java

## Jira Link
https://osi-cwds.atlassian.net/browse/INT-1517

### Tests
- [ ] I have included unit tests 
- [ ] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [x] I have NOT included tests 

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: api-core-test-support/src/main/java/gov/ca/cwds/rest/authenticate/AuthenticationUtils.java [hunk 1]
```diff
@@ -1,109 +1,94 @@
...
+  /**
+   * @param configReader - configReader
+   */
+  public AuthenticationUtils(ConfigReader configReader) {
+    if (configReader == null) {
+      this.configReader = new ConfigUtils();
+    } else {
+      this.configReader = configReader;
+    }
+  }
```
Comment (zemlyanoi) [inline api-core-test-support/src/main/java/gov/ca/cwds/rest/authenticate/AuthenticationUtils.java]: Looks like a code duplicate 

  public CwdsDevAuthenticationClient(ConfigReader configReader, String userName) {
     this.userName = userName;
+    if (configReader == null) {
+      this.configReader = new ConfigUtils();
+    } else {
+      this.configReader = configReader;
+    }
   }

File: api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigReader.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+   */
+  public CwdsAuthenticationClientConfig readConfig();
+
+  /**
+   * read the yaml values in the List.
+   * 
+   * @return the yaml list vales
+   */
+  default Yaml getYaml() {
+    Constructor constructor = new Constructor(CwdsAuthenticationClientConfig.class);
```
Comment (tparkerosi) [inline api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigReader.java]: I'm not crazy about this name. I would name it something more like clientConfig or something similar. It is about loading info for client config more than being a constructor. I think that is a more intention revealing name. If you look at the Yaml(constructor) line just below, you don't get a sense of what is going on. But Yam(clientConfig) says a lot more

File: api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigReader.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import io.dropwizard.configuration.ConfigurationSourceProvider;
+
+/**
+ * This interface is used to implement the {@link ConfigUtils}.
+ * 
+ * @author CWDS TPT-4 Team
+ *
+ */
+public interface ConfigReader extends ConfigurationSourceProvider {
```
Comment (tparkerosi) [inline api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigReader.java]: I think YmlLoader Might be a better name

File: api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigUtils.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import io.dropwizard.configuration.EnvironmentVariableSubstitutor;
+import io.dropwizard.configuration.SubstitutingSourceProvider;
+
+/**
+ * This class is used to read the yaml files values using {@link Yaml}.
+ * 
+ * @author CWDS TPT-4 Team
+ *
+ */
+public class ConfigUtils implements ConfigReader {
```
Comment (tparkerosi) [inline api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigUtils.java]: I think YmlReader, ConfigLoader, or something similar might be a better name. Its not so generic as to be doing a lot of utility stuff to the configuration. It's more generic in that it is loading a Yml configuration file.

File: api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigReader.java (hunk 3 above)
Comment (AVandana-OSI) [reply api-core-test-support/src/main/java/gov/ca/cwds/authenticate/config/ConfigReader.java]: @tparkerosi Will change the configReader to YmlLoader.

**Pull request merged**